### ISSUE TITLE
Initialize to 0

### ISSUE THREAD

**Issue by anders34** [uwhpsc-2016/homework2#49]
I saw something on the chat page that said we should initialize our matrices and arrays to zero, but do we need to do that if we're going to change them anyways? I understand the necessity if you're doing +=, but if you're not, is it really necessary?

**Comment by cswiercz** [comment]
You are correct in identifying that there is a problem with using `+=` without initialization.

Some of the `linalg.c` routines directly set values without the (direct) use of accumulators

``` c
out[i] = // some expression
```

However, other routines use

``` c
out[i] += // some expression
```

What if `out` already contained information that you didn't necessarily want there? After all, `out` is just a pointer to some location in memory and, if that location has already been used or was uninitialized and contain garbage data, then it may not necessarily be initialized to what you wanted.

This was a common error in the homework.
